### PR TITLE
SMT Subtrees can use distinct transactions

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: merkle/sparse_merkle_tree.go [hunk 1]
```diff
@@ -97,14 +99,18 @@ type subtreeWriter struct {
 
 // getOrCreateChildSubtree returns, or creates and returns, a subtree worker
```
Comment (Martin2112) [inline merkle/sparse_merkle_tree.go]: Bit pedantic but it's not actually a worker.

File: merkle/sparse_merkle_tree.go [hunk 2]
```diff
@@ -265,7 +278,11 @@ func leafQueueSize(depths []int) int {
 }
 
 // newLocalSubtreeWriter creates a new local go-routing based subtree worker.
```
Comment (Martin2112) [inline merkle/sparse_merkle_tree.go]: typo: go-routine

File: merkle/sparse_merkle_tree.go (hunk 1 above)
Comment (AlCutter) [inline merkle/sparse_merkle_tree.go]: Fixed

File: merkle/sparse_merkle_tree.go (hunk 2 above)
Comment (AlCutter) [inline merkle/sparse_merkle_tree.go]: Fixed

**Pull request merged**